experimental/devices/pca9548/pca9548.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package pca9548
+
+import (
+	"errors"
+	"strconv"
+	"sync"
+
+	"periph.io/x/periph/conn"
+	"periph.io/x/periph/conn/i2c"
+	"periph.io/x/periph/conn/i2c/i2creg"
+	"periph.io/x/periph/conn/physic"
+)
+
+// DefaultOpts is the recommended default options.
+var DefaultOpts = Opts{Address: 0x70}
+
+// Opts is the pca9548 configuration.
+type Opts struct {
+	// Address pca9548 I²C Address. Valid addresses for the NXP pca9548 are 0x70
+	// to 0x77.
+	Address int
+}
+
+// Dev is handle to a pca9548 I²C Multiplexer.
+type Dev struct {
+	// Immutable.
+	c       i2c.Bus
+	address uint16
+
+	// Mutable.
+	mu       sync.Mutex
+	port     uint8
+	numPorts uint8
+}
+
+// New creates a new handle to a pca9548 I²C multiplexer.
+func New(bus i2c.Bus, opts *Opts) (*Dev, error) {
+	d := &Dev{
+		c:       bus,
+		port:    0xFF,
+		address: uint16(opts.Address),
+		// TODO(NeuralSpaz): Make number of ports safely settable after New()
+		numPorts: 8,
+	}
+	r := make([]byte, 1)
+	err := bus.Tx(uint16(opts.Address), nil, r)
+	if err != nil {
+		return nil, errors.New("could not communicate with multiplexer: " + err.Error())
+	}
+	return d, nil
+}
+
+// Register registers port number and port name with the host. These ports can
+// then be used as any other i2c.Bus. Alias bus name that needs to be unique.
+func (d *Dev) Register(port int, alias string) error {
+	if port >= int(d.numPorts) || port < 0 {
+		return errors.New("port number must be between 0 and 7")
+	}
+
+	portID := strconv.Itoa(int(port))
+	addrStr := strconv.FormatUint(uint64(d.address), 16)
+	name := d.c.String() + "-pca9548-" + addrStr + "-" + portID
+
+	opener := newOpener(d, uint8(port), alias)
+	return i2creg.Register(name, []string{alias}, int(d.address)*10+port, opener)
+}
+
+// Halt does nothing.
+func (d *Dev) Halt() error {
+	// TODO(NeuralSpaz): Find a good way to halt that also can resume.
+	return nil
+}
+
+// String returns the bus base name for multiplexer ports.
+func (d *Dev) String() string {
+	return "pca9548-" + strconv.FormatUint(uint64(d.address), 16)
```

Comment:
Since this is going to be called frequently, I'd recommend to compute this value inside New() so here it's just `return d.name` to reduce heap churn.